src/Microsoft.Performance.SDK/Processing/IProcessingSource.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Enumerates the different sources of serialized table configuration
+    ///     data.
+    /// </summary>
+    public enum SerializationSource
+    {
+        /// <summary>
+        ///     The tables configurations are pre-built and persisted.
+        /// </summary>
+        PrebuiltTableConfiguration,
+    }
+
+    /// <summary>
+    ///     This interface is used to expose the tables associated
+    ///     with processing a given data source.
+    /// </summary>
+    public interface IProcessingSource
+    {
+        /// <summary>
+        ///     Gets the collection of tables exposed by this data source.
+        ///     These are the data tables for exposing the processed data.
+        /// </summary>
+        IEnumerable<TableDescriptor> DataTables { get; }
+
+        /// <summary>
+        ///     Gets the collection of tables that are considered to contain
+        ///     metadata about the data being processed.
+        /// </summary>
+        IEnumerable<TableDescriptor> MetadataTables { get; }
+
+        /// <summary>
+        ///     Gets the options that are supported by this data source.
+        /// </summary>
+        IEnumerable<Option> CommandLineOptions { get; }
+
+        /// <summary>
+        ///     Gets information about this <see cref="IProcessingSource"/>. This information
+        ///     is displayed in the About Box for this data source.
```

Comment:
I'm sure this is copied from the original, but these seems like a WPA-specific comment that we might want to remove or at least clarify.